File: 66 Plus One/solution.py
class Solution:
    # @param {integer[]} digits
    # @return {integer[]}
    def plusOne(self, digits):
        digits.reverse()
        nxt = 0
        nxt,digits[0] = divmod(digits[0]+1,10)
        for i in range(1,len(digits)):
            tmp = nxt
            nxt,digits[i] = divmod(digits[i]+tmp,10)
        if nxt==1:
            digits.append(1)
        digits.reverse()
        return digits
        

File: 66 Plus One/solution.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int quo = 1;
        for (int i = digits.size()-1;i>=0;i--){
            int tmp = quo;
            quo = (digits[i]+tmp)/10;
            digits[i] = (digits[i]+tmp)%10;
        }
        if (quo==1)
            digits.insert(digits.begin(),1);
        return digits;
    }
};